### PR TITLE
Cosbench workload for MCG

### PR DESCRIPTION
**Pull request by parikshithb**
COSBench is a distributed benchmark tool to test cloud object storage systems.

> Creates cosbench pods
> Prepares the workload xml 
> Submits to cosbench
> Get the results

Signed-off-by: Parikshith Byregowda <pbyregow@redhat.com>

### CODE REVIEW

File: ocs_ci/templates/workloads/cosbench/cosbench_pod.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+metadata:
+  name: cosbench-pod
+  namespace: cosbench-project
+spec:
+  securityContext:
+    fsGroup: 2000
+  serviceAccountName: default
+  containers:
+  - name: cosbench
+    image: quay.io/parikshith/noobaa  #need to push image to qe repo
```
Comment (tiffanyn108) [inline ocs_ci/templates/workloads/cosbench/cosbench_pod.yaml:12]: There is a docker image available to use: docker pull nexenta/cosbench:latest. Can we use this image?

Comment (parikshithb) [reply ocs_ci/templates/workloads/cosbench/cosbench_pod.yaml:12]: Yes, we can use from docker image but number of pulls is limited. Quay is recommended.

File: ocs_ci/ocs/cosbench.py [hunk 2]
```diff
@@ -0,0 +1,584 @@
...
+    """
+
+    def __init__(self):
+        """
+        Initializer function
+
+        """
+        self.ns_obj = OCP(kind="namespace")
+        self.namespace = constants.COSBENCH_PROJECT
+        self.configmap_obj = OCP(namespace=self.namespace, kind="ConfigMap")
```
[the remark this answers is not in the document]
Comment (parikshithb) [reply ocs_ci/ocs/cosbench.py]: done

File: tests/e2e/workloads/cosbench/test_cosbench.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    cosbench = Cosbench()
+
+    def teardown():
+        cosbench.cosbench_teardown()
+
+    request.addfinalizer(teardown)
+    return cosbench
+
+
+@tier2
```
[the remark this answers is not in the document]
Comment (parikshithb) [reply tests/e2e/workloads/cosbench/test_cosbench.py]: done

File: tests/e2e/workloads/cosbench/test_cosbench.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+class TestCosbenchWorkload(E2ETest):
```
[the remark this answers is not in the document]
Comment (parikshithb) [reply tests/e2e/workloads/cosbench/test_cosbench.py:24]: added

File: ocs_ci/ocs/cosbench.py [hunk 5]
```diff
@@ -0,0 +1,584 @@
...
+        xml_root[0].set(
+            "config",
+            f"accesskey={self.access_key_id};secretkey={self.access_key};"
+            f"endpoint={self.endpoint};path_style_access=true",
+        )
+
+    def run_init_workload(
+        self,
+        prefix,
+        containers=5,
```
Comment (shreekarSS) [inline ocs_ci/ocs/cosbench.py:112]: Can be parameterized?

File: ocs_ci/ocs/cosbench.py [hunk 6]
```diff
@@ -0,0 +1,584 @@
...
+        cosbench_pod_data["metadata"]["name"] = cosbench_pod_name
+        cosbench_pod_data["metadata"]["namespace"] = self.namespace
+        self.cosbench_pod = OCS(**cosbench_pod_data)
+        logger.info(f"Creating Cosbench pod: {self.cosbench_pod.name}")
+        self.cosbench_pod.create()
+        helpers.wait_for_resource_state(
+            resource=self.cosbench_pod, state=constants.STATUS_RUNNING, timeout=300
+        )
+        self.pod_obj.exec_oc_cmd(
+            command=f"exec {self.cosbench_pod.name} -- bash -c 'apt-get update; apt-get install -y curl'",
```
Comment (shreekarSS) [inline ocs_ci/ocs/cosbench.py:91]: docker image can be updated with curl package

File: ocs_ci/ocs/cosbench.py (hunk 5 above)
Comment (parikshithb) [reply ocs_ci/ocs/cosbench.py]: ack

File: ocs_ci/templates/workloads/cosbench/cosbench_pod.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (parikshithb) [reply ocs_ci/templates/workloads/cosbench/cosbench_pod.yaml]: Added multi arch image to ocsci repo

File: tests/e2e/workloads/cosbench/test_cosbench.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
+import logging
+import pytest
+
+from ocs_ci.framework.testlib import E2ETest, workloads
+from ocs_ci.ocs.cosbench import Cosbench
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture(scope="function")
```
Comment (shreekarSS) [inline tests/e2e/workloads/cosbench/test_cosbench.py:10]: Can be added as a fixture ?

Comment (parikshithb) [reply tests/e2e/workloads/cosbench/test_cosbench.py:10]: I will take this up in follow up PR.

**Review by shreekarSS — APPROVED** [review]
LGTM

**Review by PrasadDesala — APPROVED** [review]

**Pull request merged**